packages/f-checkout/test-utils/component-objects/f-checkout.component.js

```diff
@@ -1,23 +1,50 @@
-const checkoutComponent = () => $('[data-test-id="checkout-component"]');
+/* global browser, $ */
 
-// Form Fields
+import {
+    ALLERGEN_LINK,
+    CHECKOUT_COMPONENT,
+    FULFILLMENT_TIME_DROPDOWN,
+    USER_NOTE_INPUT,
+    GO_TO_PAYMENT_BUTTON,
+    FIELDS
+} from './f-checkout-selectors';
 
-const userNoteInput = () => $('[data-test-id="user-note"] textarea');
+const { doesElementExist } = require('../../../../test/utils/webdriverio-extensions')(browser);
 
-const allergenLink = () => $('[data-test-id="allergy-button"]');
+const checkoutComponent = () => $(CHECKOUT_COMPONENT);
 
-const fulfillmentTimeDropdown = () => $('[data-test-id="fulfillment-time"]');
+// Form Fields
 
-const goToPaymentButton = () => $('[data-test-id="confirm-payment-submit-button"]');
+const allergenLink = () => $(ALLERGEN_LINK);
+const fulfillmentTimeDropdown = () => $(FULFILLMENT_TIME_DROPDOWN);
+const goToPaymentButton = () => $(GO_TO_PAYMENT_BUTTON);
+const userNoteInput = () => $(USER_NOTE_INPUT);
 
-exports.inputs = {
-    mobileNumber: () => $('[data-test-id="input-mobile-number"]'),
-    addressLine1: () => $('[data-test-id="input-address-line-1"]'),
-    addressLine2: () => $('[data-test-id="input-address-line-2"]'),
-    addressCity: () => $('[data-test-id="input-address-city"]'),
-    addressPostcode: () => $('[data-test-id="input-address-postcode"]')
+const fields = {
+    mobileNumber: {
+        input: () => $(FIELDS.mobileNumber.input),
+        error: () => $(FIELDS.mobileNumber.error)
+    },
+    addressLine1: {
+        input: () => $(FIELDS.addressLine1.input),
+        error: () => $(FIELDS.addressLine1.error)
+    },
+    addressLine2: {
+        input: () => $(FIELDS.addressLine2.input),
+        error: ''
+    },
+    addressCity: {
+        input: () => $(FIELDS.addressCity.input),
+        error: () => $(FIELDS.addressCity.error)
+    },
+    addressPostcode: {
+        input: () => $(FIELDS.addressPostcode.input),
+        error: () => $(FIELDS.addressPostcode.error)
+    }
 };
 
+exports.isFieldErrorDisplayed = fieldName => fields[fieldName].error().isDisplayed();
+exports.isFieldDisplayed = fieldName => fields[fieldName].input().isDisplayed();
 exports.waitForCheckoutComponent = () => checkoutComponent().waitForExist();
```

Comment:
Wondering whether we should always have waitForExists() as the first function - more chronological